test/test_mount.rs

```diff
@@ -203,6 +204,31 @@ exit 23";
         assert_eq!(buf, SCRIPT_CONTENTS);
     }
 
+    pub(crate) fn test_open_tree() {
+        let tempdir = tempfile::tempdir().unwrap();
+        let abs_path = tempdir.path();
+        let res = open_tree(
+            Option::<OwnedFd>::None,
+            abs_path,
+            OpenTreeFlags::empty(),
+        );
+        let mount_fd = match res {
+            Ok(fd) => fd,
+            Err(e) if e == Errno::ENOSYS => {
+                let stderr = io::stderr();
+                let mut handle = stderr.lock();
+                writeln!(
+                    handle,
+                    "Detected kernel without `open_tree` syscall, skipping test."
+                )
+                .unwrap();
+                return;
+            }
+            Err(e) => panic!("{}", e),
+        };
+        nix::unistd::close(mount_fd.into_raw_fd()).unwrap()
```

Comment:
Do we need to manually close this returned fd, `OwnedFd` will do this for us when dropped I guess

Reply:
No, usually we don't need to. In this case, this is a test and I wanted to explicitly check the result of calling `close(mountfd)`, because currently I don't have any better ways of exercising the FD for a sanity-check.